lib/core/dispatchRequest.js

```diff
@@ -30,6 +30,7 @@ module.exports = function dispatchRequest(config) {
   config.data = transformData(
     config.data,
     config.headers,
+    config,
```

Comment:
@chinesedfan Should we just pass the `config.responseType` parameter? And also need to update [README.md](https://github.com/axios/axios/blob/master/README.md#request-config).

```javascript
  transformRequest: [function (data, headers) {
    // Do whatever you want to transform the data

    return data;
  }],
```

Reply:
> Should we just pass the config.responseType parameter?

Pass `config` has more applicable scenes, eg https://github.com/axios/axios/pull/842